crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -1,86 +1,119 @@
 //! Module for establishing Websocket-based Interfaces,
 //! i.e. those configured with `InterfaceDriver::Websocket`
 
-use super::error::InterfaceError;
 use super::error::InterfaceResult;
 use crate::conductor::interface::*;
-use crate::conductor::manager::ManagedTaskResult;
 use crate::conductor::manager::TaskManagerClient;
-use futures::FutureExt;
 use holochain_serialized_bytes::SerializedBytes;
 use holochain_types::signal::Signal;
-use holochain_websocket::ListenerHandle;
-use holochain_websocket::ListenerItem;
+use holochain_websocket::ReceiveMessage;
 use holochain_websocket::WebsocketConfig;
 use holochain_websocket::WebsocketListener;
-use holochain_websocket::WebsocketMessage;
 use holochain_websocket::WebsocketReceiver;
 use holochain_websocket::WebsocketSender;
-use std::convert::TryFrom;
 
 use std::sync::Arc;
 use tokio::sync::broadcast;
 use tokio::task::JoinHandle;
-use tokio_stream::StreamExt;
 use tracing::*;
-use url2::url2;
+
+/// Concurrency count for websocket message processing.
+/// This could represent a significant memory investment for
+/// e.g. app installations, but we also need enough buffer
+/// to accomodate interdependent operations.
+const CONCURRENCY_COUNT: usize = 128;
 
 // TODO: This is arbitrary, choose reasonable size.
+// ERROR TODO XXX (david.b): There is no such thing as backpressure in
+//                           broadcast queues! It'll just start deleting
+//                           items, and giving "Lagged" errors on receivers.
 /// Number of signals in buffer before applying
 /// back pressure.
 pub(crate) const SIGNAL_BUFFER_SIZE: usize = 50;
 /// The maximum number of connections allowed to the admin interface
 pub const MAX_CONNECTIONS: usize = 400;
 
+/// For some reason, the "task manager" doesn't actually shut down tasks...
```

Comment:
This looks like a todo too

Reply:
Task motel is @maackle 's code, I feel like altering that is outside the scope of holochain. I'm happy with the workaround here.

Reply:
Yes, certainly, I didn't mean a todo for you for this PR, but a TODO in the code for some day.

Reply:
@neonphog one downside to this workaround is that it'll cause nested `spawns`, since task_motel passes the future specified in `add_task` to a tokio JoinSet, which causes the future to be spawned. If that's not a performance issue, then this seems fine. 

As for why task_motel doesn't abort futures, the honest reason is that I didn't know futures were abortable at the time I wrote it, so everything is in terms of awaiting the `stop` signal. But another reason is that in general, the return value of the task determines subsequent behavior, and I wanted the task to be able to decide what value to return when it is stopped. In your workaround though you're using an "ignored" task so the return value doesn't matter in that case, so I agree that aborting is slightly more convenient than writing the task to `select!` on awaiting the `stop` signal (which you also could have done).

You also would have been welcome to make a change to task_motel or to ask me to, the repo is in the holochain org.

Reply:
Ok, switched to "tokio::select!" and mentioned in the comment that if task motel is updated we can remove the workaround.

Reply:
@neonphog that looks good. And actually now that I take a closer look, I think you could make this even less awkward by modifying the function in the TaskManager to do exactly this: https://github.com/holochain/holochain/blob/9a8390f789fee98616c49b235fec1751025073f1/crates/holochain/src/conductor/manager/mod.rs#L322-L328

You could change that function to be the same as your ext function, including changing the fn signature to take a `FnOnce()` instead of `FnOnce(StopListener)`. I think what you wrote is a better implementation of what's already there. And then there's no need to even change task_motel.